File: src/main/java/com/github/kindrat/cassandra/client/filter/condition/TableCondition.java
package com.github.kindrat.cassandra.client.filter.condition;

import java.util.Optional;
import java.util.Set;

import static com.github.kindrat.cassandra.client.filter.Combiner.AND;
import static com.github.kindrat.cassandra.client.filter.Combiner.OR;
import static com.github.kindrat.cassandra.client.util.StringUtil.lastWord;
import static org.apache.commons.lang3.StringUtils.equalsAnyIgnoreCase;

public class TableCondition implements StateCondition {
    @Override
    public boolean isCurrentState(String[] words, Set<String> columnNames) {
        Optional<String> lastWord = lastWord(words);
        return !lastWord.isPresent() || equalsAnyIgnoreCase(lastWord.get(), AND.name(), OR.name());
    }

    @Override
    public State name() {
        return State.TABLE;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/condition/TableCondition.kt
package com.github.kindrat.cassandra.client.filter.condition

import com.github.kindrat.cassandra.client.filter.Combiner
import com.github.kindrat.cassandra.client.util.StringUtil
import org.apache.commons.lang3.StringUtils.equalsAnyIgnoreCase

class TableCondition : StateCondition {

    override fun isCurrentState(words: Array<String>, columnNames: Set<String>): Boolean {
        val lastWord = StringUtil.lastWord(words)
        return lastWord?.let { word: String -> equalsAnyIgnoreCase(word, Combiner.AND.name, Combiner.OR.name) } ?: true
    }

    override fun name(): StateCondition.State {
        return StateCondition.State.TABLE
    }
}